packages/flutter/lib/src/services/message_codecs.dart

```diff
@@ -0,0 +1,421 @@
+  //   type of the value.
+  // * The serialization of the value itself follows the type byte.
+  // * Lengths and sizes of serialized parts are encoded using an expanding
+  //   format optimized for the common case of small non-negative integers:
+  //   * values 0..<254 using one byte with that value;
+  //   * values 254..<2^16 using three bytes, the first of which is 254, the
+  //     next two the usual big-endian unsigned representation of the value;
+  //   * values 2^16..<2^32 using five bytes, the first of which is 255, the
+  //     next four the usual big-endian unsigned representation of the value.
+  // * null, true, and false have empty serialization; they are encoded directly
+  //   in the type byte (using _kNull, _kTrue, _kFalse)
+  // * Integers representable in 32 bits are encoded using 4 bytes big-endian,
+  //   two's complement representation.
+  // * Larger integers representable in 64 bits are encoded using 8 bytes
+  //   big-endian, two's complement representation.
+  // * Still larger integers are encoded using their hexadecimal string
+  //   representation. First the length of that is encoded in the expanding
+  //   format, then follows the UTF-8 representation of the hex string.
+  // * doubles are encoded using the IEEE 754 64-bit double-precision binary
+  //   format.
+  // * Strings are encoded using their UTF-8 representation. First the length
+  //   of that in bytes is encoded using the expanding format, then follows the
+  //   UTF-8 encoding itself.
+  // * Uint8Lists, Int32Lists, Int64Lists, and Float64Lists are encoded by first
+  //   encoding the list's element count in the expanding format, then the
+  //   encoding of the list elements themselves, end-to-end with no additional
+  //   type information, using big-endian two's complement or IEEE 754 as
+  //   applicable.
+  // * Lists are encoded by first encoding their length in the expanding format,
+  //   then follows the recursive encoding of each element value, including the
+  //   type byte (Lists are assumed to be heterogeneous).
+  // * Maps are encoded by first encoding their length in the expanding format,
+  //   then follows the recursive encoding of each key/value pair, including the
+  //   type byte for both (Maps are assumed to be heterogeneous).
+  static const int _kNull = 0;
+  static const int _kTrue = 1;
+  static const int _kFalse = 2;
+  static const int _kInt32 = 3;
+  static const int _kInt64 = 4;
+  static const int _kLargeInt = 5;
+  static const int _kFloat64 = 6;
+  static const int _kString = 7;
+  static const int _kUint8List = 8;
+  static const int _kInt32List = 9;
+  static const int _kInt64List = 10;
+  static const int _kFloat64List = 11;
+  static const int _kList = 12;
+  static const int _kMap = 13;
+
+  const StandardMessageCodec();
+
+  @override
+  ByteData encodeMessage(dynamic message) {
+    if (message == null) {
+      return null;
+    }
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-braces-for-one-line-long-control-structure-statements